File: UnityScript/Example/Assets/OpenGUI/Scripts/OGProgressBar.js
#pragma strict

public class OGProgressBar extends OGWidget {
	public var value : float = 0.5;
	public var padding : Vector2;

	public function SetValue ( value : float ) {
		if ( this.value == value ) { return; }
		
		this.value = value;
		value = Mathf.Clamp ( value, 0, 1 );

		SetDirty();
	}

	private function GetThumbRect () : Rect {
		return new Rect ( drawRct.x + padding.x, drawRct.y + padding.y, ( drawRct.width - ( padding.x * 2 ) ) * value, drawRct.height - ( padding.y*2 ) );
	}
		
	
	/////////////////
	// Clean up
	/////////////////
	override function ClearChildren () {
		for ( var i : int = 0; i < transform.childCount; i++ ) {
			DestroyImmediate ( transform.GetChild ( i ).gameObject );
		}
	}

	
	/////////////////
	// Draw
	/////////////////
	override function DrawSkin () {
		// Background
		OGDrawHelper.DrawSlicedSprite ( drawRct, styles.basic.coordinates, styles.basic.border, drawDepth );

		// Thumb
		OGDrawHelper.DrawSprite ( GetThumbRect (), styles.thumb.coordinates, drawDepth ); 
	}


	/////////////////
	// Update
	/////////////////
	override function UpdateWidget () {
		// Update data
	}
}

File: OpenGUI/Assets/Plugins/OpenGUI/Scripts/OGProgressBar.cs
using UnityEngine;
using System.Collections;

public class OGProgressBar : OGWidget {
	public enum LayoutType {
		Horizontal,
		Vertical
	}

	public float value = 0.5f;
	public Vector2 padding;
	public LayoutType type;

	public void SetValue ( float value ) {
		if ( this.value == value ) { return; }

		this.value = value;
		value = Mathf.Clamp ( value, 0, 1 );
	}

	private Rect GetThumbRect () {
		if ( type == LayoutType.Horizontal ) {
			return new Rect ( drawRct.x + padding.x, drawRct.y + padding.y, ( drawRct.width - ( padding.x * 2 ) ) * value, drawRct.height - ( padding.y * 2 ) );
		} else {
			return new Rect ( drawRct.x + padding.x, drawRct.y + padding.y, ( drawRct.width - ( padding.x * 2 ) ), (drawRct.height - ( padding.y * 2 )) * value );

		}
	}


	/////////////////
	// Draw
	/////////////////
	override public void DrawSkin () {
		// Background
		OGDrawHelper.DrawSlicedSprite ( drawRct, styles.basic, drawDepth, tint, clipTo );

		// Thumb
		OGDrawHelper.DrawSprite ( GetThumbRect (), styles.thumb, drawDepth, tint, clipTo );
	}
}
